KEP 4960: Container Stop Signals (#49857)
kubernetes/website: content/en/docs/concepts/containers/container-lifecycle-hooks.md

A more detailed description of the termination behavior can be found in
[Termination of Pods](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination).

> `StopSignal`

Reviewer: This isn't a _hook_; it's a separate part of Pod lifecycle configuration.

(I'm disagreeing with the KEP, here, which calls it a hook; however, a _hook_ such as `preStop` is very different from specifying a signal to be emitted).

I recommend you add a new detail within https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination

Mention that even if you don't specify a stop signal within the Pod manifest:
- the container runtime attempts to signal graceful termination
- the signalling mechanism is OS-dependent

and then, _do_ mention that you can customise the signal for Linux Pods, by adding a `StopSignal` entry within `lifecycle:`

---

The https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/ page should link to Pod termination (but it already does, so no special change needed). The Pod termination page should also link to https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/ (but it already does, so no special change needed there either).

The StopSignal hook can be used to define a stop signal which would be sent to the container when it is

> stopped. This is equivalent to the `STOPSIGNAL` instruction that we configure when building container images.

Reviewer: Prepend “For Linux Pods“ here.

> stopped. This is equivalent to the `STOPSIGNAL` instruction that we configure when building container images.

Reviewer: Is this correct?
```suggestion
stopped. If you set this, it overrides any `STOPSIGNAL` instruction defined within the container image.
```
